tests/unit_tests/device_setup_plans/test_utils.py

```diff
@@ -37,3 +48,50 @@ def my_plan():
     eiger.async_stage.assert_called_once()
 
     eiger.disarm_detector.assert_called_once()
+
+
+@pytest.fixture()
+def null_plan():
+    return bps.null()
+
+
+def test_given_shutter_open_fails_then_eiger_disarmed_and_correct_exception_returned(
+    mock_eiger, null_plan
+):
+    detector_motion = MagicMock()
+    RE = RunEngine()
+    status = Status(done=True, success=False)
+    detector_motion.z.set = MagicMock(return_value=status)
+
+    with pytest.raises(FailedStatus) as e:
+        RE(
+            start_preparing_data_collection_then_do_plan(
+                mock_eiger, detector_motion, 100, null_plan
+            )
+        )
+    assert e.value.args[0] is status
+
+    mock_eiger.async_stage.assert_called_once()
+    detector_motion.z.set.assert_called_once()
+    mock_eiger.disarm_detector.assert_called_once()
+
+
+def test_given_detector_move_fails_then_eiger_disarmed_and_correct_exception_returned(
+    mock_eiger, null_plan
+):
+    detector_motion = MagicMock()
+    RE = RunEngine()
+    detector_motion.shutter.set = MagicMock(
+        return_value=Status(done=True, success=False)
+    )
+
+    with pytest.raises(FailedStatus):
+        RE(
+            start_preparing_data_collection_then_do_plan(
+                mock_eiger, detector_motion, 100, null_plan
+            )
+        )
+
```

Comment:
Might as well have the `assert e.value.args[0] is status` part in this test as well